### PR TITLE
Ssh shell

### PR DESCRIPTION
**Pull request by arturrez**
resolve https://github.com/ava-labs/avalanche-cli/issues/1164
host can be specified by IP, cloudID and avago NodeID
```
❯ ./bin/avalanche node status test2
Checking if node(s) are bootstrapped to Primary Network ...
Checking if node(s) are healthy ...
Getting avalanchego version of node(s)

STATUS FOR CLUSTER: test2
=========================

+---------------------+------------------------------------------+----------------+---------+---------------+------------------+-----------+
|      CLOUD ID       |                 NODE ID                  |       IP       | NETWORK | AVAGO VERSION | PRIMARY NETWORK  |  HEALTHY  |
+---------------------+------------------------------------------+----------------+---------+---------------+------------------+-----------+
| i-0cf66297dbf7c0bfb | NodeID-9wdKQ3KJU3GqvgFTc4CUYvmefEFe8t6ka | 52.201.239.113 | Fuji    | v1.10.17      | NOT_BOOTSTRAPPED | UNHEALTHY |
+---------------------+------------------------------------------+----------------+---------+---------------+------------------+-----------+
| i-0225fc39626b1edd3 | NodeID-4UnfijdUKqxkXM9wjrmSpofYm9pLqRPzY | 54.159.59.123  | Fuji    | v1.10.17      | NOT_BOOTSTRAPPED | UNHEALTHY |
+---------------------+------------------------------------------+----------------+---------+---------------+------------------+-----------+
❯ ./bin/avalanche node ssh
Cluster "test2" (Fuji)
ssh -o IdentitiesOnly=yes -o StrictHostKeyChecking=no ubuntu@52.201.239.113 -i /Users/artur/.ssh/artur-us-east-1-avalanche-cli-us-east-1-kp.pem
ssh -o IdentitiesOnly=yes -o StrictHostKeyChecking=no ubuntu@54.159.59.123 -i /Users/artur/.ssh/artur-us-east-1-avalanche-cli-us-east-1-kp.pem

❯ ./bin/avalanche node ssh test2
Cluster "test2" (Fuji)
ssh -o IdentitiesOnly=yes -o StrictHostKeyChecking=no ubuntu@52.201.239.113 -i /Users/artur/.ssh/artur-us-east-1-avalanche-cli-us-east-1-kp.pem
ssh -o IdentitiesOnly=yes -o StrictHostKeyChecking=no ubuntu@54.159.59.123 -i /Users/artur/.ssh/artur-us-east-1-avalanche-cli-us-east-1-kp.pem

❯ ./bin/avalanche node ssh test2 w
Warning: Permanently added '52.201.239.113' (ED25519) to the list of known hosts.
 21:59:27 up 7 min,  0 users,  load average: 0.95, 0.98, 0.50
USER     TTY      FROM             LOGIN@   IDLE   JCPU   PCPU WHAT
Warning: Permanently added '54.159.59.123' (ED25519) to the list of known hosts.
 21:59:28 up 7 min,  0 users,  load average: 1.61, 1.23, 0.62
USER     TTY      FROM             LOGIN@   IDLE   JCPU   PCPU WHAT

❯ ./bin/avalanche node ssh test3
Error: cluster or node test3 not found

❯ ./bin/avalanche node ssh i-0225fc39626b1edd3 w
Warning: Permanently added '54.159.59.123' (ED25519) to the list of known hosts.
 22:00:00 up 7 min,  0 users,  load average: 2.30, 1.46, 0.72
USER     TTY      FROM             LOGIN@   IDLE   JCPU   PCPU WHAT

❯ ./bin/avalanche node ssh NodeID-9wdKQ3KJU3GqvgFTc4CUYvmefEFe8t6ka w
Warning: Permanently added '52.201.239.113' (ED25519) to the list of known hosts.
 22:00:18 up 8 min,  0 users,  load average: 1.50, 1.12, 0.58
USER     TTY      FROM             LOGIN@   IDLE   JCPU   PCPU WHAT

❯ ./bin/avalanche node ssh 54.159.59.123 w
Warning: Permanently added '54.159.59.123' (ED25519) to the list of known hosts.
 22:00:35 up 8 min,  0 users,  load average: 2.58, 1.60, 0.80
USER     TTY      FROM             LOGIN@   IDLE   JCPU   PCPU WHAT
```

### CODE REVIEW

**Comment by felipemadero** [comment]
> resolve #1164 host can be specified by IP, cloudID and avago NodeID
> 
> ```
> ./bin/avalanche node shell test2 i-0d83f518a6b1a8cf3
> ./bin/avalanche node shell test2 NodeID-8KadMyBzpjVVt2hR651ZF4xd5mCv6i6as
> ./bin/avalanche node shell test2 54.84.122.219
> ```

I was expecting for ssh command to have this capability. Not another command.
After the cluster name, an index if you want to access node number n of the node list of the cluster.
If not, a node id, etc etc.

**Comment by arturrez** [comment]
> I was expecting for ssh command to have this capability. Not another command.
> After the cluster name, an index if you want to access node number n of the node list of the cluster.
> If not, a node id, etc etc.

I don't think it's possible to use index. As we use maps  order is not deterministic

**Comment by arturrez** [comment]
> > resolve #1164 host can be specified by IP, cloudID and avago NodeID
> > ```
> > ./bin/avalanche node shell test2 i-0d83f518a6b1a8cf3
> > ./bin/avalanche node shell test2 NodeID-8KadMyBzpjVVt2hR651ZF4xd5mCv6i6as
> > ./bin/avalanche node shell test2 54.84.122.219
> > ```
> 
> I was expecting for ssh command to have this capability. Not another command. After the cluster name, an index if you want to access node number n of the node list of the cluster. If not, a node id, etc etc.

File: cmd/nodecmd/shell.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	return cmd
+}
+
+func shellNode(_ *cobra.Command, args []string) error {
+	if len(args) < 2 {
+		return errors.New("invalid number of arguments")
+	}
+	clusterName := args[0]
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/shell.go:40]: we should have clustername to be consistent with `ssh` command plus https://github.com/ava-labs/avalanche-cli/pull/1297#issuecomment-1854357131

**Comment by arturrez** [comment]
pushed a commit to move shell functionality into existing ssh command

**Review by felipemadero — APPROVED** [review]
LGTM

**Pull request merged**